### DISCUSSION TITLE
how to pass variable vaw file to whisper

### DISCUSSION THREAD

**Discussion by trappedinspacetime** [openai/whisper#2006] [Q&A]
I'm trying to code a voice assistant. After hotword detected I record 3 seconds of a 16000khz mono wav 

                      sample_rate = 16000
                      duration = 5  # Adjust the duration as needed
          
                      # Record audio
                      audio_data = sd.rec(int(sample_rate * duration), samplerate=sample_rate, channels=1, dtype=np.int16)
                      sd.wait()
          
                      # Convert audio data to a BytesIO object
                      audio_bytesio = io.BytesIO()
                      audio_bytesio.write(audio_data.tobytes())
                      audio_bytesio.seek(0)
          

I don't know how to pass variable vaw file to `whisper

**Comment by phineas-pta** [comment]
wrong repo

**Reply by trappedinspacetime** [reply]
@phineas-pta Thank you for responding. Omitting this part:

                  whispercpp_command = [
                      "whispercpp",
                      "-m", "/home/kenn/Desktop/2022-10/whisper.cpp/models/ggml-base.bin",
                      "-nt",
                      "-l", "tr",
                      "-f", "-",
                      "-otxt"
                  ]
                  
How can I do that in whisper?

**Reply by phineas-pta** [reply]
search for other discusion posts containing `BytesIO`, there's already plenty
